src/sentry/api/serializers/models/organization_member/expand/roles.py

```diff
@@ -0,0 +1,65 @@
+from __future__ import annotations
+
+from typing import Any, Iterable, Mapping, Sequence, cast
+
+from sentry import features
+from sentry.api.serializers import serialize
+from sentry.api.serializers.models.user import DetailedUserSerializer
+from sentry.models import OrganizationMember, User
+from sentry.roles import organization_roles, team_roles
+from sentry.roles.manager import OrganizationRole, Role
+
+from ...role import OrganizationRoleSerializer, TeamRoleSerializer
+from .. import OrganizationMemberWithTeamsSerializer
+from ..response import OrganizationMemberWithRolesResponse
+
+
+def is_retired_role_hidden(role: OrganizationRole, member: OrganizationMember) -> bool:
```

Comment:
We can make this private-by-convention (i.e., rename to `_is_retired_role_hidden`).